### PR TITLE
add menu search

### PR DESCRIPTION
**Pull request by pavpka**

### CODE REVIEW

File: src/app/app.filter.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+@Pipe({
+    name: 'filterSearch'
+})
+
+export class FilterSearch implements PipeTransform {
+    transform(item: string, search: string) {
+        if (item && search) {
+            const contains = new RegExp(search, 'gi');
+            if (contains.test(item)) {
+                return true;
```
Comment (k-kutepov) [inline src/app/app.filter.pipe.ts]: Minor note: can we simplify this chunk with "return contains.test(item)" instead if-statement?

File: src/app/app.highlight.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    transform(item: string, search: string) {
+        if (search && item) {
+            const contains = new RegExp(search, 'gi');
+            if (contains.test(item)) {
+                return item.replace(contains, elem => `<span class='highlight'>${elem}</span>`);
+            }
+        } else {
+            return item;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (pavpka) [reply src/app/app.highlight.pipe.ts]: It looks great. However the "search" value can still be empty, so what about add check of "search":
`return (contains.test(item) && search)
...` ?

[the remark this answers is not in the document]
Comment (k-kutepov) [reply src/app/app.highlight.pipe.ts]: Indeed, you're right. I like this approach.

File: src/app/app.highlight.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { Pipe, PipeTransform } from '@angular/core';
```
Comment (klumba12) [inline src/app/app.highlight.pipe.ts]: please use highlight pipe it already exists

File: src/app/app.component.html [hunk 4]
```diff
@@ -1,24 +1,31 @@
...
+				 [fixedInViewport]="mobileQuery.matches"
+				 fixedTopGap="56">
+		<mat-form-field class="search">
+			<input matInput
+				   #searchInput
+				   placeholder="Search"
+				   type="search" />
+		</mat-form-field>
 		<mat-nav-list>
 			<ng-container *ngFor="let example of examples">
```
Comment (klumba12) [inline src/app/app.component.html]: please add fillter here `examples | filter: search.value`

File: src/app/app.component.scss [hunk 5]
```diff
@@ -37,4 +37,20 @@ h1.app-name {
 }
 .active {
     background-color: #efefef;
-}
\ No newline at end of file
+}
+.search {
+    margin: 15px;
+    width: 80%;
```
Comment (klumba12) [inline src/app/app.component.scss]: make width 100% and adjust maring if needed

File: src/app/app.component.scss [hunk 6]
```diff
@@ -37,4 +37,20 @@ h1.app-name {
...
+    display: block;
+     input{
+        height: 25px;
```
Comment (klumba12) [inline src/app/app.component.scss]: why we need this?

File: src/app/app.component.scss [hunk 7]
```diff
@@ -37,4 +37,20 @@ h1.app-name {
...
+    }
+}
+.mat-sidenav {
+   width: 260px;
```
Comment (klumba12) [inline src/app/app.component.scss]: why?

**Pull request merged**